src/main/java/com/epam/preprod/sirenko/command/impl/CreateNewProductCommand.java

```diff
@@ -0,0 +1,55 @@
+package com.epam.preprod.sirenko.command.impl;
+
+import com.epam.preprod.sirenko.AutoGeneratedProductImpl;
+import com.epam.preprod.sirenko.CreatedByInputProductImpl;
+import com.epam.preprod.sirenko.Strategy;
+import com.epam.preprod.sirenko.command.Command;
+import com.epam.preprod.sirenko.containers.FactoryContainer;
+import com.epam.preprod.sirenko.services.ProductService;
+import com.epam.preprod.sirenko.util.ConsoleReader;
+import com.epam.preprod.sirenko.util.PrintToConsole;
+import com.epam.preprod.sirenko.util.ValidatorOfConsoleInput;
+
+import java.io.IOException;
+
+public class CreateNewProductCommand implements Command {
+	private static final String CHOOSE_STRATEGY = "To create product print 1, to autogenerate product print 0";
+	private static final String CHOOSE_CATEGORY = "Print category name end press Enter";
+	private static final String INCORRECT_INPUT_CATEGORY_NAME = "No such category, try again";
+	private static final String INCORRECT_INPUT_ONE_OR_ZERO = "Please print only 0 or 1";
+	private ProductService productService;
+	
+	public CreateNewProductCommand(ProductService productService) {
+		this.productService = productService;
+	}
+	
+	@Override
+	public void execute() {
+		FactoryContainer factoryContainer = new FactoryContainer();
+		PrintToConsole.printSet(factoryContainer.getCategoriesList());
+		PrintToConsole.printString(CHOOSE_CATEGORY);
+		try {
+			String inputCategory = ConsoleReader.readFromConsole();
+			if (!factoryContainer.getCategoriesList().contains(inputCategory)) {
+				PrintToConsole.printString(INCORRECT_INPUT_CATEGORY_NAME);
+				return;
+			}
+			PrintToConsole.printString(CHOOSE_STRATEGY);
+			String inputStrategy = ConsoleReader.readFromConsole();
+			if (!ValidatorOfConsoleInput.checkInputStringIsNumberOneOrZero(inputStrategy)) {
```

Comment:
Select the option for product creation strategy once when starting the application. You should no do it during first product creation. Also we should not change product creation strategy, but currently we can

Reply:
changed